common/libs/wallet/ledger.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+import Ledger3 from 'vendor/ledger3';
+import LedgerEth from 'vendor/ledger-eth';
+import EthTx from 'ethereumjs-tx';
+import { addHexPrefix, rlp } from 'ethereumjs-util';
+import DeterministicWallet from './deterministic';
+import type { RawTransaction } from 'libs/transaction';
+import u2f from '/vendor/u2f-api';
+
+export default class LedgerWallet extends DeterministicWallet {
+  ledger: any;
+  ethApp: any;
+
+  constructor(address: string, dPath: string, index: number) {
+    super(address, dPath, index);
+    this.ledger = new Ledger3('w0w');
+    this.ethApp = new LedgerEth(this.ledger);
+  }
+
+  // modeled after
+  // https://github.com/kvhnuke/etherwallet/blob/3f7ff809e5d02d7ea47db559adaca1c930025e24/app/scripts/uiFuncs.js#L58
+  signRawTransaction(rawTx: RawTransaction): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const eTx = new EthTx(rawTx);
+
+      eTx.raw[6] = Buffer.from([rawTx.chainId]);
+      eTx.raw[7] = 0;
+      eTx.raw[8] = 0;
```

Comment:
I see this is the same as uiFuncs, but I think a better way to do this is:
```
const eTx = new EthTx({
  ...rawTx,
  v: Buffer.from([rawTx.chainId]),
  r: 0,
  s: 0
});
```
to make 6, 7, and 8 less mysterious.

Reply:
Yep, I like this a ton better! It's much more clear.